Replace UNSAFE_componentWillReceiveProps with getDerivedStateFromProps

Newbie012 opened the pull request:
Based on https://github.com/ankeetmaini/react-infinite-scroll-component/pull/178

jasonchilcott commented:
even after making these changes, chrome still gives me the same error about UNSAFE_componentWillReceiveProps.  very odd.

Newbie012 commented:
Anybody home?

kanishk30 commented on src/index.tsx:
```diff
@@ -137,15 +139,25 @@ export default class InfiniteScroll extends Component<Props, State> {
   }
 
-  UNSAFE_componentWillReceiveProps(props: Props) {
+  componentDidUpdate(nextProps: Props) {
```
@Newbie012 - Don't you think `prevProps` would be a better name here following the official documentation?
https://reactjs.org/docs/react-component.html#componentdidupdate

Newbie012 replied:
You're absolutely right. I have changed it to prevProps.

kanishk30 commented on src/index.tsx:
```diff
@@ -137,15 +139,25 @@ export default class InfiniteScroll extends Component<Props, State> {
+    // reset when data changes
+    if (dataLengthChanged) {
+      return {
+        showLoader: false,
+        pullToRefreshThresholdBreached: false,
```
instead of writing all state variables here, wouldn’t 
``` { ... prevState,  prevDataLength: nextProps.dataLength } ``` a better thing to do? 
This way if other state variables gets added which are unrelated to `getDerivedStateFromProps`, no change would be required here.
Thoughts?

ankeetmaini commented on src/index.tsx:
```diff
@@ -137,15 +139,24 @@ export default class InfiniteScroll extends Component<Props, State> {
-    // update state when new data was sent in
-    this.setState({
-      showLoader: false,
-    });
```
If you remove this the loader will never be hidden. Were you able to test this in local?

Newbie012 commented:
Sorry for the really long delay. I was too busy. I have updated the PR so that the loader will hide after the component updates. I'm not sure why I have deleted these lines in the first place. Good catch @ankeetmaini :)

iblokhin commented:
Who can merge it?

ankeetmaini commented:
I'll do a release later today! Thanks :)
